### PR TITLE
RD-188 Use mocked templates in cypress tests where possible

### PR DESCRIPTION
**Pull request by kubama**

### CODE REVIEW

**Review by qooban — COMMENTED** [review]
Looks good. Minor comments added.

One general comment: The big advantage of that work is that we will have faster and more accurate test. That's great and I'm all for that. On the other side I am a bit afraid that we are going too far from actual system tests in favor of integration (or even unit - widget) tests. That way we may lose testing typical use cases in real environment (standard use cases with default template). 

Maybe we could add 1 - 2 test cases for the whole standard flow. 

I see it as:
1. Login
2. Upload plugin
3. Set secrets
4. Upload blueprint
5. Deploy blueprint
6. Execute workflow
7. Check task graph, topology and logs

Another typical flow could be setting up users (roles, passwords), user groups, tenants by the administrator.

I see it as a quick validation test for typical user/admin flow.
I see it also a kind of "smoke test" (when we introduce selective test case execution, that kind of tests could be executed from our PRs to quickly validate if we didn't brake basic functionality).

What do you think?

File: cypress.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "testCaseSwitchClassnameAndName": true
   },
   "videoUploadOnPasses": false,
+  "watchForFileChanges": false,
```
[the remark this answers is not in the document]
Comment (kubama) [reply cypress.json:19]: Yes this is intentional. With WebStorm auto save functionality I find this feature really annoying. A common scenario:
1. Running single test in cypress test runner, the test fails
1. Switching to WebStorm to investigate/correct test code
1. Switching back to test runner to further investigate test log
1. Test runner automatically restarts discarding the log 👎 

AFAIK you're using WebStorm too, so I didn't think you could find this feature useful. I can revert the change.

[the remark this answers is not in the document]
Comment (qooban) [reply cypress.json:19]: Yes, I'm using WebStorm too. That's my configuration (I've got auto save turned off in many cases):
image

It works for me like that the test getting restarted only after triggering file save, so after pressing `Ctrl+S`.

I'm OK with you turning it off if you don't want to change the configuration. It's not critical for me. For consistency you probably need to change that also for Composer.

File: test/cypress/integration/widgets/deployment_create_spec.js [hunk 2]
```diff
@@ -1,198 +0,0 @@
...
-        verifyDeployBlueprintModal();
-    });
-
-    it('is available in Local Blueprints page in Blueprints widget', () => {
-        cy.visitPage('Local Blueprints');
-        searchTestBlueprintsInBlueprintsWidget();
-        cy.get(`tr#blueprintsTable_${testBlueprintId} i.icon.link.rocket`).click();
-
-        verifyDeployBlueprintModal();
-    });
```
[the remark this answers is not in the document]
Comment (kubama) [reply test/cypress/integration/widgets/deployment_create_spec.js:108]: This is split across relevant widget specs - `blueprints_spec`, `blueprint_actions_spec`, `deployment_button_spec`.

**Comment by kubama** [comment]
> What do you think?

Yes, I agree. I can add such tests.

**Review by qooban — APPROVED** [review]

**Pull request merged**